### PR TITLE
refactoring: use React.createRef replace ref function

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [x] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | refactoring: use React.createRef replace ref function |
| 🇨🇳 Chinese | refactoring: 使用 React.createRef 代替函数写法 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/affix/index.tsx [hunk 1]
```diff
@@ -119,14 +118,12 @@ class Affix extends React.Component<InternalAffixProps, AffixState> {
...
-
     this.measure();
   }
 
   componentWillUnmount() {
-    if (this.timeout) {
-      clearTimeout(this.timeout);
-      this.timeout = null;
+    if (this.timer) {
+      clearTimeout(this.timer);
```
Comment (zombieJ) [inline components/affix/index.tsx:126]: 这个 `= null` 留着，防御式编程，以免某天被拿来当条件

File: components/affix/index.tsx [hunk 2]
```diff
@@ -63,11 +62,11 @@ class Affix extends React.Component<InternalAffixProps, AffixState> {
     prevTarget: null,
   };
 
-  placeholderNode: HTMLDivElement;
+  private placeholderNode = createRef<HTMLDivElement>();
```
Comment (zombieJ) [inline components/affix/index.tsx:65]: `placeholderNodeRef` 加个 `Ref`后缀，下同

Comment (li-jia-nan) [reply components/affix/index.tsx]: done~

File: components/affix/index.tsx (hunk 1 above)
Comment (li-jia-nan) [reply components/affix/index.tsx:126]: done~

**Review by zombieJ — APPROVED** [review]

**Pull request merged**